lib/fluent/plugin/out_http.rb

```diff
@@ -100,11 +102,25 @@ def initialize
       @uri = nil
       @proxy_uri = nil
       @formatter = nil
+
+      @cache = []
+      @cache_id_mutex = Mutex.new
+      @cache_entry = Struct.new(:uri, :conn)
+    end
+
+    def close
+      super
+
+      # Close all open connections
+      @cache.each {|entry| entry.conn.finish if entry.conn&.started? }
     end
 
     def configure(conf)
       super
 
+      @cache = Array.new(actual_flush_thread_count, @cache_entry.new("", nil)) if @reuse_connections
+      @cache_id = 0
+
```

Comment:
```suggestion
```

We should not use the struct with `nil` or empty values if possible.
It makes code management difficult.